client/layout/index.jsx

```diff
@@ -265,6 +272,10 @@ class Layout extends Component {
 		) {
 			this.refreshColorScheme( prevProps.colorScheme, this.props.colorScheme );
 		}
+
+		if ( prevProps.locale !== this.props.locale ) {
```

Comment:
I haven't tested this code but it might be worth checking that `this.props.locale` is not `null` or and empty string because in other cases we found that WordPress uses `null` or and empty string for English.